storage/innobase/trx/trx0rseg.cc

```diff
@@ -493,10 +495,19 @@ static dberr_t trx_rseg_mem_restore(trx_rseg_t *rseg, mtr_t *mtr)
           trx_sys.recovered_binlog_offset= binlog_offset;
         trx_sys.recovered_binlog_is_legacy_pos= false;
       }
+    }
 #ifdef WITH_WSREP
-      trx_rseg_read_wsrep_checkpoint(rseg_hdr, trx_sys.recovered_wsrep_xid);
-#endif
+    XID tmp_xid;
+    tmp_xid.null();
+    /* Update recovered wsrep xid only if we found wsrep xid from
+       rseg header page and read xid seqno is larger than currently
+       recovered xid seqno. */
+    if (trx_rseg_read_wsrep_checkpoint(rseg_hdr, tmp_xid) &&
+        wsrep_xid_seqno(&tmp_xid) > wsrep_xid_seqno(&trx_sys.recovered_wsrep_xid))
+    {
+      trx_sys.recovered_wsrep_xid.set(&tmp_xid);
     }
+#endif
   }
```

Comment:
As far as I understand, this change is applicable to 10.6 as well. This function `trx_rseg_mem_restore()` was refactored in 0b47c126e31cddda1e94588799599e138400bcf8 (10.6). I assume that the purpose of this pull request is to ease the merging of #3121.

Reply:
I did not really test but even in 10.4 you can have node1 with default config and then start node2 with --innodb-undo-tablespaces=3

Reply:
Joiner node will not start if --innodb-undo-tablespaces is not exactly same as donor. Tested with 10.4 and clear error message is given:

2024-03-13 12:07:39 0 [ERROR] InnoDB: Expected to open innodb_undo_tablespaces=3 but was able to find only 0
2024-03-13 12:07:39 0 [ERROR] InnoDB: Plugin initialization aborted at srv0start.cc[1813] with error Generic error
2024-03-13 12:07:40 0 [Note] InnoDB: Starting shutdown...